File: src/main/java/am/adrian/dungeonkeeper/common/coords/MutableCoords.java
package am.adrian.dungeonkeeper.common.coords;

import lombok.Setter;

@Setter
public class MutableCoords implements Coords {

    private int x;
    private int y;

    @Override
    public int getX() {
        return x;
    }

    @Override
    public int getY() {
        return y;
    }

    public void incX(int delta) {
        x += delta;
    }

    public void incY(int delta) {
        y += delta;
    }
}

File: src/main/kotlin/am/adrian/dungeonkeeper/model/coords/MutableCoords.kt
package am.adrian.dungeonkeeper.model.coords

data class MutableCoords(
    private var x: Int = 0,
    private var y: Int = 0
) : Coords {

    override fun x(): Int = x

    override fun y(): Int = y

    fun x(x: Int) {
        this.x = x
    }

    fun y(y: Int) {
        this.y = y
    }

    fun incX(delta: Int) {
        x += delta
    }

    fun incY(delta: Int) {
        y += delta
    }
}
